internal/services/mongodb/snapshot.go

```diff
@@ -0,0 +1,212 @@
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time of the last update of the snapshot",
+			},
+			"expires_at": {
+				Type:             schema.TypeString,
+				Description:      "Expiration date (Format ISO 8601). Cannot be removed.",
+				Required:         true,
+				ValidateDiagFunc: verify.IsDate(),
+			},
+			"region": regional.Schema(),
+		},
+		CustomizeDiff: customdiff.All(),
+	}
+}
+
+func ResourceSnapshotCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	mongodbAPI, zone, region, err := newAPIWithZoneAndRegion(d, m)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	instanceID := locality.ExpandID(d.Get("instance_id").(string))
+	createReq := &mongodb.CreateSnapshotRequest{
+		InstanceID: instanceID,
+		Region:     region,
+		Name:       types.ExpandOrGenerateString(d.Get("name"), "snapshot"),
+		ExpiresAt:  types.ExpandTimePtr(d.Get("expires_at")),
+	}
+
+	snapshot, err := mongodbAPI.CreateSnapshot(createReq, scw.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if snapshot != nil {
+		d.SetId(zonal.NewIDString(zone, snapshot.ID))
+		_, err = waitForSnapshot(ctx, mongodbAPI, region, instanceID, snapshot.ID, d.Timeout(schema.TimeoutCreate))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	return ResourceSnapshotRead(ctx, d, m)
+}
+
+func ResourceSnapshotRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	mongodbAPI, region, err := newAPIWithRegion(d, m)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	zone, snapshotID, err := zonal.ParseID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	instanceID := locality.ExpandID(d.Get("instance_id").(string))
+	snapshot, err := waitForSnapshot(ctx, mongodbAPI, region, instanceID, snapshotID, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	_ = d.Set("id", snapshot.ID)
+	_ = d.Set("instance_id", zonal.NewIDString(zone, snapshot.InstanceID))
+	_ = d.Set("name", snapshot.Name)
+	_ = d.Set("instance_name", snapshot.InstanceName)
+	_ = d.Set("size", int64(snapshot.Size))
+	_ = d.Set("node_type", snapshot.NodeType)
+	_ = d.Set("volume_type", snapshot.VolumeType.Type)
+	_ = d.Set("expires_at", types.FlattenTime(snapshot.ExpiresAt))
+	_ = d.Set("created_at", types.FlattenTime(snapshot.CreatedAt))
+	_ = d.Set("updated_at", types.FlattenTime(snapshot.UpdatedAt))
+	_ = d.Set("region", snapshot.Region.String())
+
+	return nil
+}
+
+func ResourceSnapshotUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	mongodbAPI, region, err := newAPIWithRegion(d, m)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	_, snapshotID, err := zonal.ParseID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	updateReq := &mongodb.UpdateSnapshotRequest{
+		SnapshotID: snapshotID,
+		Region:     region,
+	}
+
+	if d.HasChange("name") {
```

Comment:
Introduce variable to trigger a change if this boolean value changed. Like `has_changed`